packages/provider/src/providers/coinbase.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  Chain,
+  ChainId,
+  EthereumProvider,
+  EthProviderRpcError,
+  ProviderProxy,
+  RawProvider,
+  TransactionResponse,
+  TxRequestBody,
+} from '@/types'
+import { HttpProvider } from 'web3-core'
+import Web3 from 'web3/types'
+import { ChainTypes, ProviderEvents } from '@/enums'
+import {
+  connectEthProvider,
+  detectCurrentEthChain,
+  getEthExplorerAddressUrl,
+  getEthExplorerTxUrl,
+  handleEthError,
+  hexToDecimal,
+  requestAddEthChain,
+  requestSwitchEthChain,
+} from '@/helpers'
+
+declare global {
+  interface Window {
+    Web3: typeof Web3
+  }
+}
+
+export class CoinbaseProvider implements ProviderProxy {
```

Comment:
let's try to fix this thing with DRY. we can create BaseEthereumProvider which do all same but without static method `get providerType` and then for coinbase and metamask we will do:

```ts
export class MetamaskProvider extends BaseEthereumProvider implements ProviderProxy  {
  constructor(provider: RawProvider) {
   super(provider)
  }

  static get providerType(): Providers {
    return Providers.Coinbase
  }
}
```

In that way we will respect DRY principle

Reply:
fixed

Reply:
fixed